### PR TITLE
NDRS-845: Added validator bonding ITS

### PR DESCRIPTION
**Pull request by darthsiroftardis**
REF: https://casperlabs.atlassian.net/browse/NDRS-845

CHANGELOG:

- Added an integration test that simulates a scenario where a sixth node in a 5 node network bonds in and joins the network
- The test will check if the sixth node is a validator by querying the auction contract and checking for the public key in the auction info

### CODE REVIEW

File: utils/nctl/sh/scenarios/bond_its.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            quiet="TRUE"
+
+    log "node-$NODE_ID auction bid submitted -> $BID_AMOUNT CSPR"
+
+    log_step "awaiting 10 seconds for auction bid deploys to finalise"
+    sleep 10.0
+}
+
+
+function get_node_public_key_hex() {
```
Comment (TomVasile) [inline utils/nctl/sh/scenarios/bond_its.sh:63]: This is available in common. Note: there are two" `get_node_public_key_hex` & `get_node_public_key_hex_extended`. There is a comment explaining the differences.

File: utils/nctl/sh/scenarios/bond_its.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+function assert_new_bonded_validator() {
+    local NODE_ID=${1}
+    local HEX=$(get_node_public_key_hex "$NODE_ID")
+    if ! $(nctl-view-chain-auction-info | grep -q "$HEX"); then
+      echo "Could not find key in bids"
+      exit 1
+    fi
+}
+
+function assert_node_proposed() {
```
Comment (TomVasile) [inline utils/nctl/sh/scenarios/bond_its.sh:80]: This is available in common too.

File: utils/nctl/sh/scenarios/accounts_toml/bond_its.accounts.toml [hunk 3]
```diff
@@ -0,0 +1,140 @@
+# FAUCET.
```
Comment (TomVasile) [inline utils/nctl/sh/scenarios/accounts_toml/bond_its.accounts.toml:1]: Just curious what changes were made to the toml that were needed for the test?

File: utils/nctl/sh/scenarios/chainspecs/bond_its.chainspec.toml.in [hunk 4]
```diff
@@ -0,0 +1,209 @@
+[protocol]
```
Comment (TomVasile) [inline utils/nctl/sh/scenarios/chainspecs/bond_its.chainspec.toml.in:1]: likewise here, can you let me know what changes you had to make here that the default cant be used? Just so i have a note to reference if I need to debug this

**Review by TomVasile — CHANGES_REQUESTED** [review]
just some small things. Also you need to add this to `ci/nightly-test.sh`

File: utils/nctl/sh/scenarios/chainspecs/bond_its.chainspec.toml.in (hunk 4 above)
Comment (darthsiroftardis) [reply utils/nctl/sh/scenarios/chainspecs/bond_its.chainspec.toml.in:1]: `maximum_round_exponent = 14` and `block_max_transfer_count = 1` are the two differences from the regular chainspec.

File: utils/nctl/sh/scenarios/accounts_toml/bond_its.accounts.toml (hunk 3 above)
Comment (darthsiroftardis) [reply utils/nctl/sh/scenarios/accounts_toml/bond_its.accounts.toml:1]: I re-purposed the same `its14.accounts.toml` used in `its14` and the difference between those two files is that the there is an entry for the sixth node has 1000x the delegation rate of the remaining nodes. This was to ensure that the node produces a block within a reasonable span of time

File: utils/nctl/sh/scenarios/bond_its.sh (hunk 1 above)
Comment (darthsiroftardis) [reply utils/nctl/sh/scenarios/bond_its.sh:63]: Removed

File: utils/nctl/sh/scenarios/bond_its.sh (hunk 2 above)
Comment (darthsiroftardis) [reply utils/nctl/sh/scenarios/bond_its.sh:80]: Removed as well

Comment (darthsiroftardis) [reply utils/nctl/sh/scenarios/bond_its.sh:80]: Is this in common, I can't find it

Comment (TomVasile) [reply utils/nctl/sh/scenarios/bond_its.sh:72]: AH, my mistake not common. its in a test though: https://github.com/CasperLabs/casper-node/blob/dev/utils/nctl/sh/scenarios/itst14.sh

Comment (TomVasile) [reply utils/nctl/sh/scenarios/bond_its.sh:72]: so can be moved to common

**Review by TomVasile — APPROVED** [review]

**Pull request merged**